File: frontend/scripts/generate-icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

// Simple icon generator using canvas
function generateIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Background
  ctx.fillStyle = '#0d9488';
  roundRect(ctx, 0, 0, size, size, size * 0.15);
  ctx.fill();
  
  // Wallet body
  const w = size * 0.55;
  const h = size * 0.38;
  const x = (size - w) / 2;
  const y = size * 0.32;
  
  ctx.fillStyle = '#ffffff';
  roundRect(ctx, x, y, w, h, size * 0.05);
  ctx.fill();
  
  // Wallet flap
  ctx.fillStyle = '#f0fdfa';
  ctx.beginPath();
  ctx.moveTo(x, y + h * 0.2);
  ctx.lineTo(x, y);
  ctx.lineTo(x + w, y);
  ctx.lineTo(x + w, y + h * 0.2);
  ctx.lineTo(x, y + h * 0.2);
  ctx.fill();
  
  // Card line
  ctx.fillStyle = 'rgba(13, 148, 136, 0.3)';
  const cardX = x + w * 0.15;
  const cardY = y + h * 0.55;
  roundRect(ctx, cardX, cardY, w * 0.6, h * 0.12, size * 0.02);
  ctx.fill();
  
  // W letter
  ctx.fillStyle = '#0d9488';
  ctx.font = `bold ${size * 0.22}px system-ui, sans-serif`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('W', size / 2, y + h + (size - y - h) / 2);
  
  return canvas.toBuffer('image/png');
}

function roundRect(ctx, x, y, w, h, r) {
  ctx.beginPath();
  ctx.moveTo(x + r, y);
  ctx.lineTo(x + w - r, y);
  ctx.quadraticCurveTo(x + w, y, x + w, y + r);
  ctx.lineTo(x + w, y + h - r);
  ctx.quadraticCurveTo(x + w, y + h, x + w - r, y + h);
  ctx.lineTo(x + r, y + h);
  ctx.quadraticCurveTo(x, y + h, x, y + h - r);
  ctx.lineTo(x, y + r);
  ctx.quadraticCurveTo(x, y, x + r, y);
  ctx.closePath();
}

function generateMaskableIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');
  
  // Full background for maskable
  ctx.fillStyle = '#0d9488';
  ctx.fillRect(0, 0, size, size);
  
  // Smaller icon in center (80% safe zone)
  const iconSize = size * 0.6;
  const offset = (size - iconSize) / 2;
  
  // Wallet body
  const w = iconSize * 0.55;
  const h = iconSize * 0.38;
  const x = offset + (iconSize - w) / 2;
  const y = offset + iconSize * 0.25;
  
  ctx.fillStyle = '#ffffff';
  roundRect(ctx, x, y, w, h, iconSize * 0.05);
  ctx.fill();
  
  // W letter
  ctx.fillStyle = '#0d9488';
  ctx.font = `bold ${iconSize * 0.22}px system-ui, sans-serif`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('W', size / 2, y + h + (offset + iconSize - y - h) / 2);
  
  return canvas.toBuffer('image/png');
}

const publicDir = path.join(__dirname, '..', 'public');

// Generate icons
[192, 512].forEach(size => {
  try {
    const buffer = generateIcon(size);
    fs.writeFileSync(path.join(publicDir, `icon-${size}.png`));
    console.log(`Generated icon-${size}.png`);
  } catch (e) {
    console.log(`canvas not available, creating placeholder`);
  }
});

try {
  const maskableBuffer = generateMaskableIcon(512);
  fs.writeFileSync(path.join(publicDir, 'icon-512-maskable.png'));
  console.log('Generated icon-512-maskable.png');
} catch (e) {
  console.log('canvas not available');
}

File: frontend/scripts/generate_icons.py
"""
Generate PWA icons from wally-logo.svg using Pillow.
Creates: icon-192.png, icon-512.png, icon-512-maskable.png
"""

import os
from PIL import Image, ImageDraw, ImageFont

def draw_rounded_rect(draw, xy, radius, fill):
    x0, y0, x1, y1 = xy
    draw.rectangle([x0 + radius, y0, x1 - radius, y1], fill=fill)
    draw.rectangle([x0, y0 + radius, x1, y1 - radius], fill=fill)
    draw.pieslice([x0, y0, x0 + 2*radius, y0 + 2*radius], 180, 270, fill=fill)
    draw.pieslice([x1 - 2*radius, y0, x1, y0 + 2*radius], 270, 360, fill=fill)
    draw.pieslice([x0, y1 - 2*radius, x0 + 2*radius, y1], 90, 180, fill=fill)
    draw.pieslice([x1 - 2*radius, y1 - 2*radius, x1, y1], 0, 90, fill=fill)

def generate_icon(size, output_path, maskable=False):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    bg_radius = int(size * 0.15)
    
    if maskable:
        # Maskable: full background, icon centered
        bg_color = (13, 148, 136)  # Teal-500
        draw.rectangle([0, 0, size, size], fill=bg_color)
        icon_scale = 0.65
    else:
        # Standard: rounded rect background
        bg_color = (13, 148, 136)
        draw_rounded_rect(draw, (0, 0, size - 1, size - 1), bg_radius, bg_color)
        icon_scale = 0.75
    
    offset = int(size * (1 - icon_scale) / 2)
    icon_size = int(size * icon_scale)
    
    # Wallet body
    wallet_w = int(icon_size * 0.55)
    wallet_h = int(icon_size * 0.38)
    wallet_x = offset + (icon_size - wallet_w) // 2
    wallet_y = offset + int(icon_size * 0.18)
    wallet_radius = int(wallet_w * 0.08)
    
    # Wallet body (main)
    draw_rounded_rect(draw, (wallet_x, wallet_y, wallet_x + wallet_w, wallet_y + wallet_h), wallet_radius, (255, 255, 255))
    
    # Wallet flap (top portion)
    flap_h = int(wallet_h * 0.35)
    draw_rounded_rect(draw, (wallet_x, wallet_y, wallet_x + wallet_w, wallet_y + flap_h), wallet_radius, (240, 253, 250))
    
    # Card slot
    card_w = int(wallet_w * 0.55)
    card_h = int(wallet_h * 0.12)
    card_x = wallet_x + (wallet_w - card_w) // 2
    card_y = wallet_y + int(wallet_h * 0.5)
    draw_rounded_rect(draw, (card_x, card_y, card_x + card_w, card_y + card_h), int(card_h * 0.3), (13, 148, 136))
    
    # W letter
    try:
        font_size = int(icon_size * 0.24)
        font = ImageFont.truetype("arial.ttf", font_size)
    except (IOError, OSError):
        try:
            font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", font_size)
        except (IOError, OSError):
            font = ImageFont.load_default()
    
    text = "W"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_w = bbox[2] - bbox[0]
    text_h = bbox[3] - bbox[1]
    text_x = (size - text_w) // 2
    text_y = wallet_y + wallet_h + (offset + icon_size - wallet_y - wallet_h - text_h) // 2
    draw.text((text_x, text_y), text, fill=(13, 148, 136), font=font)
    
    img.save(output_path, "PNG")
    print(f"Generated: {output_path} ({size}x{size})")

if __name__ == "__main__":
    public_dir = os.path.join(os.path.dirname(__file__), "..", "public")
    public_dir = os.path.abspath(public_dir)
    
    generate_icon(192, os.path.join(public_dir, "icon-192.png"))
    generate_icon(512, os.path.join(public_dir, "icon-512.png"))
    generate_icon(512, os.path.join(public_dir, "icon-512-maskable.png"), maskable=True)
    
    print("All icons generated!")
